producer-c/producer-cloudwatch-integ/KvsProducerSampleCloudwatch.cpp

```diff
@@ -0,0 +1,205 @@
+
+#include "CanaryStreamUtils.h"
+
+volatile ATOMIC_BOOL sigCaptureInterrupt;
+
+VOID sigintHandler(INT32 sigNum)
+{
+    ATOMIC_STORE_BOOL(&sigCaptureInterrupt, TRUE);
+}
+
+// add frame pts, frame index, original frame size, CRC to beginning of buffer
+VOID addCanaryMetadataToFrameData(PFrame pFrame) {
+    PBYTE pCurPtr = pFrame->frameData;
+    putUnalignedInt64(pCurPtr, pFrame->presentationTs / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+    pCurPtr += SIZEOF(UINT64);
+    putUnalignedInt32(pCurPtr, pFrame->index);
+    pCurPtr += SIZEOF(UINT32);
+    putUnalignedInt32(pCurPtr, pFrame->size);
+    pCurPtr += SIZEOF(UINT32);
+    MEMSET(pCurPtr, 0x00, SIZEOF(UINT64));
+    putUnalignedInt64(pCurPtr, COMPUTE_CRC32(pFrame->frameData, pFrame->size));
+}
+
+VOID createCanaryFrameData(PFrame pFrame) {
+    UINT32 i;
+
+    for (i = CANARY_METADATA_SIZE; i < pFrame->size; i++) {
+        pFrame->frameData[i] = RAND();
+    }
+    addCanaryMetadataToFrameData(pFrame);
+}
+
+PCHAR getCanaryStr(UINT32 canaryType) {
+    switch (canaryType) {
+        case 0:
+            return (PCHAR)"realtime";
+        case 1:
+            return (PCHAR)"offline";
+        default:
+            return (PCHAR)"";
+    }
+}
+
+VOID adjustStreamInfoToCanaryType(PStreamInfo pStreamInfo, UINT32 canaryType) {
+    switch (canaryType) {
+        case 0:
+            pStreamInfo->streamCaps.streamingType = STREAMING_TYPE_REALTIME;
+            break;
+        case 1:
+            pStreamInfo->streamCaps.streamingType = STREAMING_TYPE_OFFLINE;
+            break;
+        default:
+            break;
+    }
+
+}
+
+INT32 main(INT32 argc, CHAR *argv[])
+{
+#ifndef _WIN32
+    signal(SIGINT, sigintHandler);
+#endif
+    SET_INSTRUMENTED_ALLOCATORS();
+    PDeviceInfo pDeviceInfo = NULL;
+    PStreamInfo pStreamInfo = NULL;
+    PStreamCallbacks pStreamCallbacks = NULL;
+    PClientCallbacks pClientCallbacks = NULL;
+    CLIENT_HANDLE clientHandle = INVALID_CLIENT_HANDLE_VALUE;
+    STREAM_HANDLE streamHandle = INVALID_STREAM_HANDLE_VALUE;
+    STATUS retStatus = STATUS_SUCCESS;
+    PCHAR accessKey = NULL, secretKey = NULL, sessionToken = NULL, streamNamePrefix = NULL, canaryTypeStr = NULL, region = NULL, cacertPath = NULL;
+    CHAR streamName[MAX_STREAM_NAME_LEN + 1];
+    Frame frame;
+    UINT32 frameIndex = 0, fileIndex = 0, canaryType = 0;
+    UINT64 fragmentSizeInByte = 0;
+    UINT64 lastKeyFrameTimestamp = 0;
+    PCanaryStreamCallbacks pCanaryStreamCallbacks = NULL;
+
+    SRAND(time(0));
+    Aws::SDKOptions options;
```

Comment:
We might also need to initialize the endianness by calling initializeEndianness() API

Reply:
Added.